### DISCUSSION TITLE
It would be great if it was open source

### DISCUSSION THREAD

**Discussion by mg459046365** [skiptools/skip#286] [Q&A] [answered]
It would be great if it was open source (if not, I understand), and it would also promote its development and improvement. Of course, if it could get official support from Apple, like Vapor, it would be perfect. I like Swift very much, and what you are doing is very meaningful.

**Comment by marcprux** [comment, marked as answer]
Thank you for your support and positive words.

All of our runtime libraries, like [skip-foundation](https://github.com/skiptools/skip-foundation) and [skip-ui](https://github.com/skiptools/skip-ui), are indeed open-source, as is the [Skip plugin](https://github.com/skiptools/skip) itself. This means that you can have confidence that your app projects will continue to be buildable even if we were to disappear tomorrow, and that you would be able to continue to examine and improve the source code.

The only part that is proprietary is the Skip transpiler implementation itself, which is the tool that the plugin uses to perform the transpilation from Swift to Kotlin. This is the value that you get from the Skip Developer Program.
